engine/src/test/scala/cromwell/engine/workflow/WorkflowDockerLookupActorSpec.scala

```diff
@@ -0,0 +1,391 @@
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+
+
+class WorkflowDockerLookupActorSpec extends TestKitSuite("WorkflowDockerLookupActorSpecSystem") with FlatSpecLike with Matchers with ImplicitSender with BeforeAndAfter with Mockito {
+  it should "wait and resubmit the docker request when it gets a backpressure message" in {
+    val dockerValue = "ubuntu:latest"
+    val dockerId = DockerImageIdentifier.fromString(dockerValue).get.asInstanceOf[DockerImageIdentifierWithoutHash]
+
+    val workflowId = WorkflowId.randomId()
+    val dockerHashingActor = TestProbe()
+    val backpressureWaitTime = 2 seconds
+
+    val lookupActor = TestActorRef(Props(new TestWorkflowDockerLookupActor(workflowId, dockerHashingActor.ref, StartNewWorkflow, backpressureWaitTime)), self)
+    val request = DockerHashRequest(dockerId)
+    lookupActor ! request
+
+    dockerHashingActor.expectMsg(request)
+    dockerHashingActor.reply(BackPressure(request))
+    // Give a couple of seconds of margin to account for test latency etc...
+    dockerHashingActor.expectMsg(backpressureWaitTime.+(2 seconds), request)
+  }
+
+  it should "not look up the same tag again after a successful lookup" in {
+    val dockerValue = "ubuntu:latest"
+    val dockerId = DockerImageIdentifier.fromString(dockerValue).get.asInstanceOf[DockerImageIdentifierWithoutHash]
+
+    val workflowId = WorkflowId.randomId()
+    val dockerHashingActor = TestProbe()
+
+    val databaseConfig = ConfigFactory.load.getConfig("database")
+
+    var numWrites = 0
+
+    val databaseInterface = new SlickDatabase(databaseConfig) {
+
+      override def queryDockerHashStoreEntries(workflowExecutionUuid: String)
+                                              (implicit ec: ExecutionContext): Future[Seq[DockerHashStoreEntry]] =
+        throw new RuntimeException("should not be called, this workflow is not restarting")
+
+      override def addDockerHashStoreEntry(dockerHashStoreEntry: DockerHashStoreEntry)(implicit ec: ExecutionContext): Future[Unit] = {
+        numWrites = numWrites + 1
+        Future.successful(())
+      }
+    }.initialized
+
+    val lookupActor = TestActorRef(WorkflowDockerLookupActor.props(workflowId, dockerHashingActor.ref, StartNewWorkflow, databaseInterface))
+    val request = DockerHashRequest(dockerId)
+    lookupActor ! request
+    val response = DockerHashSuccessResponse(DockerHashResult("md5", "AAAAAAAA"), request)
+
+    // The WorkflowDockerLookupActor should not have the hash for this tag yet and will need to query the dockerHashingActor.
+    dockerHashingActor.expectMsg(request)
+    dockerHashingActor.reply(response)
+    // The WorkflowDockerLookupActor should forward the success message to this actor.
+    expectMsg(response)
+    numWrites should equal(1)
+
+    // Now the WorkflowDockerLookupActor should now have this hash in its mappings and should not query the dockerHashingActor again.
+    dockerHashingActor.expectNoMsg()
```

Comment:
shouldn't this be one line below, after we send a second request to the `lookupActor` ?